src/nunavut/lang/c/__init__.py

```diff
@@ -206,12 +208,65 @@ def filter_macrofy(value: str) -> str:
 
         jinja_filter_tester(filter_macrofy, template, rendered, 'c')
 
+    Note that individual tokens are not stropped so the appearance of an identifier
+    in the ``SCREAMING_SNAKE_CASE`` output my be different then the token as it appears
+    on its own. For example:
+
+    .. code-block:: python
+
+        # "register" is reserved so it will be stropped if it appears as an
+        # identifier...
+        template = '''#ifndef {{ "namespaced.Type.register" | macrofy }}
+        {{ "register" | id }}
+        '''
+
+        # ...but it will not be stropped within the macro.
+        rendered = '''#ifndef NAMESPACED_TYPE_REGISTER
+        _register'''
+
+    .. invisible-code-block: python
+
+        jinja_filter_tester([filter_macrofy, filter_id], template, rendered, 'c')
+
+    If stropping is enabled, however, the entire token generated by this filter will be stropped:
+
+    .. code-block:: python
+
+        # Given
+        template = '#ifndef {{ "_starts_with_underscore" | macrofy }}'
+
+        # then
+        rendered = '#ifndef _sTARTS_WITH_UNDERSCORE'
+
+    .. invisible-code-block: python
+
+        jinja_filter_tester(filter_macrofy, template, rendered, 'c')
+
+    And again with stropping disabled:
+
+    .. code-block:: python
+
+        # Given
+        template = '#ifndef {{ "_starts_with_underscore" | macrofy }}'
+
+        # then with stropping disabled
+        rendered = '#ifndef _STARTS_WITH_UNDERSCORE'
+
+    .. invisible-code-block: python
+
+        config_overrides = {'nunavut.lang.c': {'enable_stropping': False }}
+        lctx = configurable_language_context_factory(config_overrides, 'c')
+        jinja_filter_tester(filter_macrofy, template, rendered, lctx)
 
     :param str value: The value to transform.
 
     :returns: A valid C preprocessor identifier token.
     """
-    return value.replace(' ', '_').replace('.', '_').upper()
+    macrofied_value = filter_to_screaming_snake_case(str(value))
```

Comment:
Should we not apply consistent naming throughout? Instead of

```
#define UAVCAN_TIME_TIME_SYSTEM_0_1_MONOTONIC_SINCE_BOOT 0
```

have

```
#define uavcan_time_TimeSystem_0_1_MONOTONIC_SINCE_BOOT 0
```

I see that some other entities have been renamed already as suggested during the last review but the constants stand out still.

Reply:
The constants are macros as such there is a consistency that all macros are SCREAMING_SNAKE_CASE to communicate to the developer an important difference in their nature. I believe this is the only deviation from the modified snake case used elsewhere.

Reply:
Whatever we choose it will end up looking odd -- either macros that don't look like macros or macros that don't look like the things they are supposed to represent. I am inclined to break the coding convention rather than DSDL convention here because I think it adds clarity and removes the risk (which is admittedly marginal though) of case-sensitive name conflicts.

Also, consider the issues with metric suffixes. We discussed it a while ago on the forum and the convention made its way into the specification: if an attribute has a metric suffix, the case of the suffix is independent of the case of the attribute itself. For example, there may be `MAX_ENERGY_mJ` where `mJ` stands for millijoule. Do you see how the screaming snake case transformation might introduce an error of six orders of magnitude? Scary :snake: